src/AutoRest.CSharp/Common/Output/Models/Types/SystemObjectType.cs

```diff
@@ -0,0 +1,181 @@
+        {
+            var type = _type;
+            while (type is not null)
+            {
+                foreach (var property in type.GetProperties())
+                {
+                    yield return property;
+                }
+                type = type.BaseType;
+            }
+        }
+
+        protected override ObjectTypeDiscriminator? BuildDiscriminator()
+        {
+            return null;
+        }
+
+        public string GetNameWithoutGeneric(Type t)
+        {
+            if (!t.IsGenericType)
+                return t.Name;
+
+            string name = t.Name;
+            int index = name.IndexOf('`');
+            return index == -1 ? name : name.Substring(0, index);
+        }
+
+        protected override ObjectTypeConstructor BuildInitializationConstructor()
+        {
+            MemberDeclarationOptions memberDeclarationOptions = new MemberDeclarationOptions("protected", DefaultName, _csharpType);
+            List<Parameter> parameters = new List<Parameter>();
+            foreach (var param in GetInitializationParameters())
+            {
+                parameters.Add(new Parameter(ToCamelCase(param.Name!), $"The {param.Name}", GetTypeFromSystem(param.ParameterType), null, false));
+
+            }
+
+            List<ObjectPropertyInitializer> initializers = new List<ObjectPropertyInitializer>();
+            foreach (var autoRestProperty in Properties)
+            {
+                Reference reference = new Reference(ToCamelCase(autoRestProperty.Declaration.Name), autoRestProperty.ValueType);
+                initializers.Add(new ObjectPropertyInitializer(autoRestProperty, reference));
+            }
+
+            ObjectTypeConstructor ctor = new ObjectTypeConstructor(memberDeclarationOptions, parameters.ToArray(), initializers.ToArray(), GetBaseCtor());
+            return ctor;
+        }
+
+        protected override IEnumerable<ObjectTypeProperty> BuildProperties()
+        {
+            foreach (var property in _myProperties)
+            {
+                var getter = property.GetGetMethod();
+                var setter = property.GetSetMethod();
+                MemberDeclarationOptions memberDeclarationOptions = new MemberDeclarationOptions(
+                    getter is not null && getter.IsPublic ? "public" : "internal",
+                    property.Name,
+                    GetTypeFromSystem(property.PropertyType));
+                Property prop = new Property();
+                prop.Nullable = false;
+                prop.ReadOnly = GetReadOnly(property); //TODO read this from attribute from reference object
+                prop.SerializedName = ToCamelCase(property.Name);
+                prop.Summary = $"Gets{GetPropertySummary(setter)} {property.Name}";
+                prop.Required = true;
+                if (property.PropertyType.IsGenericType && property.PropertyType.GetGenericTypeDefinition() == typeof(IDictionary<,>))
```

Comment:
Can we add comment here to capture the node on the just enough type supported for Mgmt.